src/optimizations/push_filter_before_sort.rs

```diff
@@ -0,0 +1,27 @@
+use crate::{pattern, workflow::WorkflowStep};
+
+use super::{Group, Optimization, Pattern};
+
+pub struct PushFilterBeforeSort;
+
+impl Optimization for PushFilterBeforeSort {
+    fn pattern(&self) -> Pattern {
+        pattern!(Scan Sort Filter)
```

Comment:
remove scan and then it will apply also for queries like `scan | filter | sort | filter` to be `scan | filter | filter | sort` for example